axelor-redmine/src/main/java/com/axelor/apps/redmine/service/imports/projects/RedmineImportProjectServiceImpl.java

```diff
@@ -200,31 +174,63 @@ public void importProject(
       }
 
       if (!updatedOnMap.isEmpty()) {
-        String values =
-            updatedOnMap.entrySet().stream()
-                .map(
-                    entry ->
-                        "("
-                            + entry.getKey()
-                            + ",TO_TIMESTAMP('"
-                            + entry.getValue()
-                            + "', 'YYYY-MM-DD HH24:MI:SS'))")
-                .collect(Collectors.joining(","));
-
-        String query =
-            String.format(
-                "UPDATE project_project as project SET updated_on = v.updated_on from (values %s) as v(id,updated_on) where project.id = v.id",
-                values);
-
-        JPA.em().createNativeQuery(query).executeUpdate();
+        executeProjectUpdatingQuery();
       }
     }
 
     String resultStr =
         String.format("Redmine Project -> ABS Project : Success: %d Fail: %d", success, fail);
     setResult(getResult() + String.format("%s \n", resultStr));
     LOG.debug(resultStr);
-    success = fail = 0;
+    success = 0;
+    fail = 0;
+  }
+
+  private void fillFieldMapWithImportMappingList() {
+    List<RedmineImportMapping> redmineImportMappingList =
+        redmineImportMappingRepository
+            .all()
+            .filter(
+                "self.redmineImportConfig.redmineMappingFieldSelect in (?1, ?2)",
+                RedmineImportConfigRepository.MAPPING_FIELD_PROJECT_TRACKER,
+                RedmineImportConfigRepository.MAPPING_FIELD_VERSION_STATUS)
+            .fetch();
+
+    for (RedmineImportMapping redmineImportMapping : redmineImportMappingList) {
+      fieldMap.put(redmineImportMapping.getRedmineValue(), redmineImportMapping.getOsValue());
+    }
+  }
+
+  private void fillSelectionMapWithSelectionList() {
+    ArrayList<Option> selectionList =
+        new ArrayList<>(MetaStore.getSelectionList("support.project.version.status.select"));
+    ResourceBundle fr = I18n.getBundle(Locale.FRANCE);
+    ResourceBundle en = I18n.getBundle(Locale.ENGLISH);
+
+    for (Option option : selectionList) {
+      selectionMap.put(fr.getString(option.getTitle()), Integer.parseInt(option.getValue()));
+      selectionMap.put(en.getString(option.getTitle()), Integer.parseInt(option.getValue()));
+    }
+  }
+
+  public void executeProjectUpdatingQuery() {
+    String values =
+        updatedOnMap.entrySet().stream()
+            .map(
+                entry ->
+                    "("
+                        + entry.getKey()
+                        + ",TO_TIMESTAMP('"
+                        + entry.getValue()
+                        + "', 'YYYY-MM-DD HH24:MI:SS'))")
+            .collect(Collectors.joining(","));
+
+    String query =
+        String.format(
+            "UPDATE project_project as project SET updated_on = v.updated_on from (values %s) as v(id,updated_on) where project.id = v.id",
+            values);
+
+    JPA.em().createNativeQuery(query).executeUpdate();
```

Comment:
Since we're refactoring this module, it would be a great idea to use JPQL queries in order to not be limited by a single database technology (here, psql)